src/pyflex/window_selector.py

```diff
@@ -505,6 +543,45 @@ def calculate_ttimes(self):
         self.ttimes = sorted(tts, key=lambda x: x["time"])
         logger.info("Calculated travel times.")
 
+    def reject_on_select_mode(self):
+        """
+        Reject based on select mode. Will reject windows outside of the wave
+        category specified. For example, if self.config.select_mode == "body_only",
+        only body wave windows will be selected.
+        """
+        dist_in_km = geodetics.calcVincentyInverse(
+            self.station.latitude, self.station.longitude, self.event.latitude,
+            self.event.longitude)[0] / 1000.0
+
+        offset = self.event.origin_time - self.observed.stats.starttime
+
+        select_mode = self.config.select_mode
+        if select_mode == "all_waves":
+            min_time = self.ttimes[0]["time"] - self.config.min_period + offset
+            max_time = self.observed.stats.endtime - self.observed.stats.starttime
+        elif select_mode == "body_and_surface":
+            min_time = self.ttimes[0]["time"] - self.config.min_period + offset
+            max_time = dist_in_km / self.config.min_surface_wave_velocity + \
+                    self.config.min_period + offset
+        elif select_mode == "body_wave":
+            min_time = self.ttimes[0]["time"] - self.config.min_period + offset
+            max_time = dist_in_km / self.config.max_surface_wave_velocity + \
+                    3 * self.config.min_period + offset
```

Comment:
Personal choice... haha :) I think people may get confuse when seeing this magic number, "3".
What would be the better solution? Put that config?